cpp/src/arrow/c/bridge.cc

```diff
@@ -1977,6 +1978,24 @@ Result<std::shared_ptr<Array>> ImportDeviceArray(struct ArrowDeviceArray* array,
   return ImportDeviceArray(array, *maybe_type, mapper);
 }
 
+Result<std::shared_ptr<MemoryManager>> DefaultDeviceMapper(ArrowDeviceType device_type,
+                                                           int64_t device_id) {
+  if (device_type != ARROW_DEVICE_CPU) {
+    return Status::NotImplemented("Only importing data on CPU is supported");
+  }
+  return default_cpu_memory_manager();
+}
+
+Result<std::shared_ptr<Array>> ImportDeviceArray(struct ArrowDeviceArray* array,
+                                                 std::shared_ptr<DataType> type) {
+  return ImportDeviceArray(array, type, DefaultDeviceMapper);
+}
```

Comment:
Do we want to provide such an API that uses a default DeviceMapper? 

With the current APIs here, I assume the idea is that it's the responsibility of the user (i.e. the library or application using Arrow C++ to consume data through the C Device interface) to provide the device mapping as they see fit. 
In the case of exposing this in pyarrow, it's pyarrow that is the user of those APIs and I think pyarrow certainly wants to have a default mapping provided (not to be specified by the user of pyarrow). In theory I could write this `DefaultDeviceMapper` function in cython to keep this on the pyarrow side, but this might also be useful for other users of the C++ APIs?

(I suppose when we add a default in C++, I could also give the existing signatures a default parameter value for `mapper`, instead of adding those two additional signatures)

cc @zeroshade @pitrou

Reply:
> Do we want to provide such an API that uses a default DeviceMapper?

Yes, that sounds reasonable to me. I think that in many (most?) cases, users will want to use whatever device mapper is registered for the given device type.

Reply:
Also:

> I could also give the existing signatures a default parameter value for `mapper`, instead of adding those two additional signatures

Yes, that would reduce the proliferation of different functions. You could simply have something like `const DeviceMemoryMapper& mapper = {}`.

Reply:
The difficulty with providing a default device mapper here is that it created a circular dependency due to the `ArrowDeviceType` being defined in `abi.h` and required linking against `libarrow_cuda.so`

Reply:
That might be a reason to keep this default on the pyarrow side? (we can implement the mapper function in C++, but only provide it as the default argument on the Python side)

In Python, we can more easily dynamically check if pyarrow.cuda module is available, and if so provide a different default mapper (that includes GPU devices).